contracts/BoostedMasterChefJoe.sol

```diff
@@ -145,149 +170,101 @@ contract BoostedMasterChefJoe is Initializable, OwnableUpgradeable, ReentrancyGu
         emit Init(balance);
     }
 
-    /// @notice Returns the number of BMCJ pools.
-    /// @return pools The amount of pools in this farm
-    function poolLength() external view returns (uint256 pools) {
-        pools = poolInfo.length;
-    }
-
     /// @notice Add a new LP to the pool. Can only be called by the owner.
     /// @param _allocPoint AP of the new pool.
+    /// @param _veJoeShareBp Share of rewards allocated in proportion to user's liquidity
+    /// and veJoe balance
     /// @param _lpToken Address of the LP ERC-20 token.
     /// @param _rewarder Address of the rewarder delegate.
     function add(
-        uint256 _allocPoint,
+        uint96 _allocPoint,
+        uint32 _veJoeShareBp,
         IERC20 _lpToken,
         IRewarder _rewarder
     ) external onlyOwner {
-        require(!lpTokens.contains(address(_lpToken)), "BoostedMasterChefJoe: LP already added");
+        require(
+            !checkPoolDuplicate[_lpToken],
+            "BoostedMasterChefJoe: LP already added"
+        );
+        require(
+            _veJoeShareBp <= 10_000,
+            "BoostedMasterChefJoe: veJoeShareBp needs to be lower than 10000"
+        );
         require(poolInfo.length <= 50, "BoostedMasterChefJoe: Too many pools");
```

Comment:
Add zero address validation for `_lpToken`

Reply:
It would revert with the sanity check on `lpToken.balanceOf(address(this));`, do we really want to add that requirement ? It's not really useful